libs/hwdrivers/src/rplidar/src/rplidar_driver.cpp

```diff
@@ -349,10 +570,7 @@ u_result RPlidarDriverSerialImpl::ascendScanData(rplidar_response_measurement_no
     }
 
     // all the data is invalid
-    if (i == count){
-        delete[] tmpbuffer;
-        return RESULT_OPERATION_FAIL;
-    }
+    if (i == count) return RESULT_OPERATION_FAIL;
 
     //Tune tail
     for (i = count - 1; i >= 0; i--) {
```

Comment:
In the whole code block, the logic is ok.  once  all invalid may occure you said, but it is processed at   i== cout.
`    // all the data is invalid
    if (i == count) return RESULT_OPERATION_FAIL;

    //Tune tail
    for (i = count - 1; i >= 0; i--) {
        if(nodebuffer[i].distance_q2 == 0) {
            continue;
        } else {
            while(i != (count - 1)) {
                i++;
                float expect_angle = (nodebuffer[i-1].angle_q6_checkbit >> RPLIDAR_RESP_MEASUREMENT_ANGLE_SHIFT)/64.0f + inc_origin_angle;
                if (expect_angle > 360.0f) expect_angle -= 360.0f;
                _u16 checkbit = nodebuffer[i].angle_q6_checkbit & RPLIDAR_RESP_MEASUREMENT_CHECKBIT;
                nodebuffer[i].angle_q6_checkbit = (((_u16)(expect_angle * 64.0f)) << RPLIDAR_RESP_MEASUREMENT_ANGLE_SHIFT) + checkbit;
            }
            break;
        }
    }`